src/SpecterWallet.ts

```diff
@@ -29,7 +32,7 @@ export class SpecterWallet {
   async setBitcoindClient(): Promise<string> {
     const wallets = await SpecterWallet.listWallets()
 
-    const pattern = yamlConfig.rebalancing.onchainWallet ?? "specter"
+    const pattern = this.config.onchainWallet
```

Comment:
Interesting point.

I know @bodymindarts wanted to still keep some of these concerns separate though. Could another approach be to pass global configs directly into the `SpecterWallet` constructor and then automatically call `getSpecterWalletConfig` inside the constructor?

Reply:
@dolcalmi would this sort of change work? https://github.com/GaloyMoney/galoy/pull/344/commits/7187ea636f271014d7c87ce6e02e2009f84ee57d

I also changed the `??` check to a tighter `||` check since empty strings are no longer valid.